### PR TITLE
feat: crop image when changing/adding profile picture

### PR DESCRIPTION
**Pull request by TheGuyWithoutH**
Adds the possibility for a user to crop an image for its profile picture for higher customization.

image

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/culturequest/ProfileCreatorActivityTest.java [hunk 1]
```diff
@@ -154,6 +162,24 @@ public void wrongUsernamesFailProfileCreation() throws InterruptedException {
...
+//        Intent intent = new Intent();
+//        intent.setData(Uri.parse("content://media/external/images/media/1"));
+//        Instrumentation.ActivityResult result = new Instrumentation.ActivityResult(RESULT_OK, intent);
+//        intending(hasAction(Intent.ACTION_PICK)).respondWith(result);
+//
+//        onView(withId(R.id.profile_picture)).perform(click());
+//
+//        intended(hasComponent(UCropActivity.class.getName()));
+//    }
+
```
Comment (alberttkt) [inline app/src/androidTest/java/ch/epfl/culturequest/ProfileCreatorActivityTest.java:184]: Why is this test commented? Should he be removed?

File: app/src/androidTest/java/ch/epfl/culturequest/SettingsActivityTest.java [hunk 2]
```diff
@@ -93,6 +104,25 @@ public void UpdateProfileButtonSendsBackToProfileFragmentWithCorrectUsername() {
...
+//    public void afterPictureChosenGoToCrop() {
+//        Intent intent = new Intent();
+//        intent.setData(Uri.parse("content://media/external/images/media/1"));
+//        Instrumentation.ActivityResult result = new Instrumentation.ActivityResult(RESULT_OK, intent);
+//        intending(hasAction(Intent.ACTION_PICK)).respondWith(result);
+//
+//        onView(withId(R.id.profile_picture)).perform(click());
+//
+//        intended(hasComponent(UCropActivity.class.getName()));
+//    }
```
Comment (alberttkt) [inline app/src/androidTest/java/ch/epfl/culturequest/SettingsActivityTest.java:126]: Same here?

File: app/src/main/java/ch/epfl/culturequest/SettingsActivity.java [hunk 3]
```diff
@@ -177,7 +190,35 @@ public void selectProfilePicture(View view) {
     }
 
     private void openGallery() {
-        profilePictureSelector.launch(new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI));
+        // start the gallery activity to select a picture with result code TAKE_PICTURE
+        startActivityForResult(new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI), TAKE_PICTURE);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
```
Comment (alberttkt) [inline app/src/main/java/ch/epfl/culturequest/SettingsActivity.java:198]: This method looks very similar to the one in ProfileCreationActivity, you should consider refactoring

**Review by alberttkt — CHANGES_REQUESTED** [review]
Seems good! I just have some suggestions!

File: app/src/androidTest/java/ch/epfl/culturequest/ProfileCreatorActivityTest.java (hunk 1 above)
Comment (TheGuyWithoutH) [reply app/src/androidTest/java/ch/epfl/culturequest/ProfileCreatorActivityTest.java:184]: There is a problem with the permissions that needs to be granted on the CI. I will need to investigate this issue next sprint.

But in the meantime the tests are already written.

Comment (alberttkt) [reply app/src/androidTest/java/ch/epfl/culturequest/ProfileCreatorActivityTest.java:183]: Ok I see then you can keep it that way

File: app/src/main/java/ch/epfl/culturequest/utils/CropUtils.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        options.setShowCropGrid(false);
+        options.setActiveControlsWidgetColor(ContextCompat.getColor(context, R.color.colorPrimary));
+        options.setToolbarColor(ContextCompat.getColor(context, R.color.background));
+        options.setStatusBarColor(ContextCompat.getColor(context, R.color.background));
+        options.setToolbarWidgetColor(ContextCompat.getColor(context, R.color.colorPrimary));
+        options.setToolbarTitle("Adjust your profile picture");
+        return options;
+    }
+
+    public static void manageCropFlow(int requestCode, int resultCode, Intent data, Activity activity, Function<Uri, Void> displayProfilePic, View rootView) {
```
Comment (alberttkt) [inline app/src/main/java/ch/epfl/culturequest/utils/CropUtils.java:40]: Nice Refactoring!

**Review by alberttkt — APPROVED** [review]
This PR seems ready to go, good job of refactoring

**Pull request merged**